Microsoft.Toolkit.Uwp/IncrementalLoadingCollection/IncrementalLoadingCollection.cs

```diff
@@ -0,0 +1,207 @@
+            }
+
+            private set
+            {
+                if (value != _hasMoreItems)
+                {
+                    _hasMoreItems = value;
+                    OnPropertyChanged(new PropertyChangedEventArgs(nameof(HasMoreItems)));
+                }
+            }
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="IncrementalLoadingCollection{TSource, IType}"/> class, optionally specifying how many items to load for each data page.
+        /// </summary>
+        /// <param name="itemsPerPage">
+        /// The number of items to retrieve for each call. Default is 20.
+        /// </param>
+        /// <param name="onStartLoading">
+        /// An <see cref="Action"/> that is called when a retrieval operation begins.
+        /// </param>
+        /// <param name="onEndLoading">
+        /// An <see cref="Action"/> that is called when a retrieval operation ends.
+        /// </param>
+        /// <param name="onError">
+        /// An <see cref="Action"/> that is called if an error occours during data retrieval.
+        /// </param>
+        public IncrementalLoadingCollection(int itemsPerPage = 20, Action onStartLoading = null, Action onEndLoading = null, Action<Exception> onError = null)
+        {
+            _source = new TSource();
+
+            _onStartLoading = onStartLoading;
+            _onEndLoading = onEndLoading;
+            _onError = onError;
+
+            _itemsPerPage = itemsPerPage;
+            _hasMoreItems = true;
+        }
+
+        /// <summary>
+        /// Initializes incremental loading from the view.
+        /// </summary>
+        /// <param name="count">
+        /// The number of items to load.
+        /// </param>
+        /// <returns>
+        /// An object of the <see cref="LoadMoreItemsAsync(uint)"/> that specifies how many items have been actually retrieved.
+        /// </returns>
+        public IAsyncOperation<LoadMoreItemsResult> LoadMoreItemsAsync(uint count)
+            => AsyncInfo.Run((c) => LoadMoreItemsAsync(count, c));
+
+        private async Task<LoadMoreItemsResult> LoadMoreItemsAsync(uint count, CancellationToken cancellationToken)
+        {
+            uint resultCount = 0;
+            _cancellationToken = cancellationToken;
+
+            try
+            {
+                if (!_cancellationToken.IsCancellationRequested)
+                {
+                    IEnumerable<IType> data = null;
+                    try
+                    {
+                        IsLoading = true;
+                        data = await LoadDataAsync(count, _cancellationToken);
+                    }
+                    catch (OperationCanceledException)
+                    {
+                        // The operation has been canceled using the Cancellation Token.
+                    }
+                    catch (Exception ex) when (_onError != null)
+                    {
+                        _onError.Invoke(ex);
```

Comment:
There is an Exception filter on line 167 to check whether _onError is != null, so there is no need to use the ?. operator.